src/Razor/src/Microsoft.AspNetCore.Razor.Language/Intermediate/ComponentAttributeIntermediateNode.cs

```diff
@@ -68,6 +62,40 @@ public ComponentAttributeIntermediateNode(TagHelperPropertyIntermediateNode prop
             }
         }
 
+        public ComponentAttributeIntermediateNode(TagHelperDirectiveAttributeIntermediateNode directiveAttributeNode)
+        {
+            if (directiveAttributeNode == null)
+            {
+                throw new ArgumentNullException(nameof(directiveAttributeNode));
+            }
+
+            var attributeName = directiveAttributeNode.AttributeName;
+            if (attributeName.StartsWith("@"))
```

Comment:
Thinking this might be good to do at the `TagHelperDirectiveAttributeIntermediateNode` lowering layer instead of here. Assuming it makes sense that is.

Reply:
Same comment as above.